openmetadata-spec/src/main/resources/json/schema/security/credentials/gcpValues.json

```diff
@@ -36,6 +36,26 @@
         }
       ]
     },
+    "audience": {
+      "title": "Audience",
+      "description": "Google Security Token Service audience which contains the resource name for the workload identity pool and the provider identifier in that pool.",
+      "type": "string"
+    },
+    "subjectTokenType": {
+      "title": "Subject Token Type",
+      "description": "Google Security Token Service subject token type based on the OAuth 2.0 token exchange spec.",
+      "type": "string"
+    },
+    "tokenURL": {
+      "title": "Token URL",
+      "description": "Google Security Token Service token exchange endpoint.",
+      "type": "string"
+    },
+    "credentialSource": {
+      "title": "Credential Source",
+      "description": "This object defines the mechanism used to retrieve the external credential from the local environment so that it can be exchanged for a GCP access token via the STS endpoint",
+      "type": "object"
```

Comment:
Could you please share an example of this value? Trying to understand why this is an `object`

Reply:
The following is an example of a json whose type is external_account, which allows authentication between AWS and GCP. Check credential_source to determine the necessary logic to retrieve the external credential which should be used to construct the subject token to pass to the STS endpoint. 
please check detail from https://google.aip.dev/auth/4117.
```
{
  "type": "external_account",
  "audience": "//iam.googleapis.com/projects/GCP_PROJECT_ID/locations/global/workloadIdentityPools/openmetadata/providers/aws-provider",
  "subject_token_type": "urn:ietf:params:aws:token-type:aws4_request",
  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/SERVICE_ACCOUNT_EMAIL:generateAccessToken",
  "token_url": "https://sts.googleapis.com/v1/token",
  "credential_source": {
    "environment_id": "aws1",
    "region_url": "http://YOUR_AWS_ENDPOINT/latest/meta-data/placement/availability-zone",
    "url": "http://YOUR_AWS_ENDPOINT/latest/meta-data/iam/security-credentials",
    "regional_cred_verification_url": "https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15"
  }
}
```

Reply:
Thanks @TrsNium, then I think we might need a couple of updates to keep things organized on the JSON Schemas.

1. Looks like the values you added `audience`, `subjectTokenType `, `tokenURL` & `credentialSource` are only used when working with `type: external_account`. Makes me wonder if it'd make sense to keep two separate definitions, the one we had for `service_account` and your new one for `external_account` as `oneOf` property. [Example](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/datalakeConnection.json#L31).
2. Regarding `credential_source`, if we're expecting a predefined set of parameters, it should be a new typed definition, not just an `object`. Otherwise it might be hard for users to understand what needs to be added there and it would render strange on the UI.

Thanks

Reply:
@pmbrull @TrsNium as seen in the docs, i see that `credential_source` can have multiple different parameters based on the source ( AWS, Azure, etc.. ) as it looks like there are 4 different credential_sources with 4 different parameters